cmd/nerdctl/load.go

```diff
@@ -56,106 +45,35 @@ func newLoadCommand() *cobra.Command {
 	return loadCommand
 }
 
-func loadAction(cmd *cobra.Command, _ []string) error {
-	in := cmd.InOrStdin()
+func processLoadCommandFlags(cmd *cobra.Command) (types.LoadCommandOptions, error) {
 	input, err := cmd.Flags().GetString("input")
 	if err != nil {
-		return err
+		return types.LoadCommandOptions{}, err
 	}
 	globalOptions, err := processRootCmdFlags(cmd)
 	if err != nil {
-		return err
-	}
-	if input != "" {
-		f, err := os.Open(input)
-		if err != nil {
-			return err
-		}
-		defer f.Close()
-		in = f
-	} else {
-		// check if stdin is empty.
-		stdinStat, err := os.Stdin.Stat()
-		if err != nil {
-			return err
-		}
-		if stdinStat.Size() == 0 && (stdinStat.Mode()&os.ModeNamedPipe) == 0 {
-			return errors.New("stdin is empty and input flag is not specified")
-		}
-	}
-	decompressor, err := compression.DecompressStream(in)
-	if err != nil {
-		return err
+		return types.LoadCommandOptions{}, err
 	}
-
 	allPlatforms, err := cmd.Flags().GetBool("all-platforms")
 	if err != nil {
-		return err
+		return types.LoadCommandOptions{}, err
 	}
 	platform, err := cmd.Flags().GetStringSlice("platform")
 	if err != nil {
-		return err
+		return types.LoadCommandOptions{}, err
 	}
-	platMC, err := platformutil.NewMatchComparer(allPlatforms, platform)
-	if err != nil {
-		return err
-	}
-
-	return loadImage(decompressor, cmd, globalOptions, platMC, false)
+	return types.LoadCommandOptions{
+		GOptions:     *globalOptions,
```

Comment:
no change required here, but we might gradually change the pointer usage of global option to non-pointer to avoid unintented shared memory among goroutines (https://github.com/containerd/nerdctl/pull/1795#discussion_r1064256900) and also be consistent with other command option usages.

1. Those global options in function signatures should be fine, as they should be gradually replaced by individual command options (with global option being a field).
2. `processRootCmdFlags` should be changed to return a struct not a struct pointer. Similarly, SubCommandOptions should hold a global option struct (as in `LoadCommandOptions`) not a struct pointer.

I guess the original intent of using global option as a pointer is to avoid unnecessary copies, but I think it should be fine as those option structs are quite small, and also pointers use heap allocation which means more GC work.